custom_components/unifiprotect/button.py

```diff
@@ -0,0 +1,62 @@
+"""Support for Ubiquiti's Unifi Protect NVR."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from pyunifiprotect.api import ProtectApiClient
+from pyunifiprotect.data.base import ProtectAdoptableDeviceModel
+
+from custom_components.unifiprotect.data import UnifiProtectData
+
+from .const import DEVICES_WITH_ENTITIES, DOMAIN
+from .entity import UnifiProtectEntity
+from .models import UnifiProtectEntryData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Discover devices on a UniFi Protect NVR."""
+    entry_data: UnifiProtectEntryData = hass.data[DOMAIN][entry.entry_id]
+    protect = entry_data.protect
+    protect_data = entry_data.protect_data
+
+    async_add_entities(
+        [
+            UnifiProtectMediaPlayer(
+                protect,
+                protect_data,
+                device,
+            )
+            for device in protect_data.get_by_types(DEVICES_WITH_ENTITIES)
+        ]
+    )
+
+
+class UnifiProtectMediaPlayer(UnifiProtectEntity, ButtonEntity):
+    """A Ubiquiti UniFi Protect Reboot button."""
+
+    def __init__(
+        self,
+        protect: ProtectApiClient,
+        protect_data: UnifiProtectData,
+        device: ProtectAdoptableDeviceModel,
+    ):
+        """Initialize an Unifi camera."""
+
+        super().__init__(protect, protect_data, device, None)
+
+        self._attr_name = f"{self.device.name} Reboot Device"
+        self._attr_entity_registry_enabled_default = False
```

Comment:
Not sure if you are aware, but they added device classes for buttons. (example: https://github.com/home-assistant/core/pull/60625/files). Probably not released yet, however maybe good to take into account for a follow up PR / this PR.

```device_class=ButtonDeviceClass.RESTART```

Reply:
I did not know that, thanks for the heads up. I will either add it for this PR or make a note of it for the core merge so we do not need to depend on 2021.12 after I dig into it more.

Reply:
It looks button entities to begin with are 2021.12+. So, I added the device class and added a check for 2021.12 before initializing the platform. Definitely thanks for the heads up about the device class though.